components/calendar/src/chinese.rs

```diff
@@ -273,42 +274,40 @@ impl Calendar for Chinese {
                 12 => tinystr!(4, "M12"),
                 _ => tinystr!(4, "und"), // maximum num of months in a non-leap year is 12
             }
+        } else if ordinal == leap_month {
+            match ordinal {
+                1 => tinystr!(4, "und"), // cannot have a leap month before the actual month
+                2 => tinystr!(4, "M01L"),
+                3 => tinystr!(4, "M02L"),
+                4 => tinystr!(4, "M03L"),
+                5 => tinystr!(4, "M04L"),
+                6 => tinystr!(4, "M05L"),
+                7 => tinystr!(4, "M06L"),
+                8 => tinystr!(4, "M07L"),
+                9 => tinystr!(4, "M08L"),
+                10 => tinystr!(4, "M09L"),
+                11 => tinystr!(4, "M10L"),
+                12 => tinystr!(4, "M11L"),
+                13 => tinystr!(4, "M12L"),
+                _ => tinystr!(4, "und"), // maximum num of months in a leap year is 13
+            }
         } else {
-            if ordinal == leap_month {
-                match ordinal {
-                    1 => tinystr!(4, "und"), // cannot have a leap month before the actual month
-                    2 => tinystr!(4, "M01L"),
-                    3 => tinystr!(4, "M02L"),
-                    4 => tinystr!(4, "M03L"),
-                    5 => tinystr!(4, "M04L"),
-                    6 => tinystr!(4, "M05L"),
-                    7 => tinystr!(4, "M06L"),
-                    8 => tinystr!(4, "M07L"),
-                    9 => tinystr!(4, "M08L"),
-                    10 => tinystr!(4, "M09L"),
-                    11 => tinystr!(4, "M10L"),
-                    12 => tinystr!(4, "M11L"),
-                    13 => tinystr!(4, "M12L"),
-                    _ => tinystr!(4, "und"), // maximum num of months in a leap year is 13
-                }
-            } else {
-                // if ordinal > leap_month; this means the date is in a leap year
-                match ordinal {
-                    1 => tinystr!(4, "und"), // this implies the leap month is < 1, which is impossible
-                    2 => tinystr!(4, "und"), // this implies the leap month is = 1, which is impossible
-                    3 => tinystr!(4, "M02"),
-                    4 => tinystr!(4, "M03"),
-                    5 => tinystr!(4, "M04"),
-                    6 => tinystr!(4, "M05"),
-                    7 => tinystr!(4, "M06"),
-                    8 => tinystr!(4, "M07"),
-                    9 => tinystr!(4, "M08"),
-                    10 => tinystr!(4, "M09"),
-                    11 => tinystr!(4, "M10"),
-                    12 => tinystr!(4, "M11"),
-                    13 => tinystr!(4, "M12"),
-                    _ => tinystr!(4, "und"), // maximum number of months in a leap year is 13
-                }
+            // if ordinal > leap_month; this means the date is in a leap year
```

Comment:
nit: perhaps we should call the variable leap_month_ordinal so that it is clear that it's an ordinal as well